### PR TITLE
[WIP] Create common lib

### PR DESCRIPTION
**Pull request by jackgerrits**
Testing how warning CI reacts to deprecations

### CODE REVIEW

File: common/include/vowpalwabbit/common/rand48.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+constexpr int BIAS = 127 << 23;
+
+namespace VW
+{
+namespace common
+{
+inline float merand48(uint64_t& initial)
+{
+  static_assert(sizeof(int) == sizeof(float), "Floats and ints are converted between, they must be the same size.");
```
Comment (peterychang) [inline common/include/vowpalwabbit/common/rand48.h:30]: I don't see `int` playing any part in this. We're taking the lower 32 bits of a `uint64_t`, shift/masking it and casting to a float. Really, this should be checking `sizeof(uint32_t) == sizeof(float)`

Comment (jackgerrits) [reply common/include/vowpalwabbit/common/rand48.h:30]: Yes, that's true this really should be the actual type we are manipulating `int32_t`

Comment (jackgerrits) [reply common/include/vowpalwabbit/common/rand48.h:30]: I'll fix this in master since this is a draft

**Comment by jackgerrits** [comment]
Closing as this is only feasible when we are converged on CMake. Will readdress at that time.

**Pull request closed without merging**